### PR TITLE
tidy: enable misc-unused-parameters

### PR DESCRIPTION
**Pull request by Riolku**
About half of this was done by hand, and about half with an automated tool.

### CODE REVIEW

File: src/function/vector_list_functions.cpp [hunk 1]
```diff
@@ -97,7 +97,8 @@ vector_function_definitions ListCreationVectorFunction::getDefinitions() {
 }
 
 std::unique_ptr<FunctionBindData> ListRangeVectorFunction::bindFunc(
-    const binder::expression_vector& arguments, kuzu::function::FunctionDefinition* definition) {
+    const binder::expression_vector& arguments,
+    kuzu::function::FunctionDefinition* /*definition*/) {
```
Comment (acquamarin) [inline src/function/vector_list_functions.cpp:101]: Remove redundant `kuzu::function` namespace.

File: src/function/vector_list_functions.cpp [hunk 2]
```diff
@@ -235,7 +236,8 @@ vector_function_definitions ListConcatVectorFunction::getDefinitions() {
 }
 
 std::unique_ptr<FunctionBindData> ListConcatVectorFunction::bindFunc(
-    const binder::expression_vector& arguments, kuzu::function::FunctionDefinition* definition) {
+    const binder::expression_vector& arguments,
+    kuzu::function::FunctionDefinition* /*definition*/) {
```
Comment (acquamarin) [inline src/function/vector_list_functions.cpp:240]: ditto, remove redundant kuzu::function namespace.

**Review by acquamarin — APPROVED** [review]
I am not quite sure whether we should strictly follow the clang-format rule at this stage. 
Maybe @ray6080 @andyfengHKU @mewim should also comment on this.

File: src/function/vector_list_functions.cpp (hunk 1 above)
Comment (Riolku) [reply src/function/vector_list_functions.cpp]: good catch

**Comment by ray6080** [comment]
> I am not quite sure whether we should strictly follow the clang-format rule at this stage. Maybe @ray6080 @andyfengHKU @mewim should also comment on this.

I'm not against this change to enhance our `tidy` checks, though I feel we can deprioritize coding style checks for now, as this can easily add conflicts to other's local changes, and this seems like something good to be done in a bulky way. Maybe we can schedule code style cleanups before each release when we reach a state that most changes are frozen to be merged into master. Also, I'm curious if this can help moving towards `-Werror` in the longer term.

**Comment by Riolku** [comment]
I think most clang-tidy related changes should be less invasive than this one, or at least, so I hope.

#2194 was large because of me adding many tests, and renaming to `stringFormat`. #2156 was not very large.

**Comment by Riolku** [comment]
Realistically, with a team working full time, these kinds of changes will always cause mass conflicts. Even if we do it when master is frozen, this doesn't really prevent people from rebasing their branches. All it means is I have to rebase less when making this PR, maybe.

I agree that these changes can be deprioritized.

File: src/function/vector_null_functions.cpp [hunk 3]
```diff
@@ -10,6 +10,7 @@ namespace function {
 void VectorNullFunction::bindExecFunction(ExpressionType expressionType,
     const binder::expression_vector& children, scalar_exec_func& func) {
     assert(children.size() == 1);
+    (void)children;
```
Comment (andyfengHKU) [inline src/function/vector_null_functions.cpp:13]: let's remove assertion and comment out the parameter?

File: src/function/vector_struct_functions.cpp [hunk 4]
```diff
@@ -100,6 +100,7 @@ std::unique_ptr<FunctionBindData> StructExtractVectorFunctions::bindFunc(
     }
     auto key = ((binder::LiteralExpression&)*arguments[1]).getValue()->getValue<std::string>();
     assert(definition->returnTypeID == LogicalTypeID::ANY);
+    (void)definition;
```
Comment (andyfengHKU) [inline src/function/vector_struct_functions.cpp:103]: ditto

**Review by andyfengHKU — APPROVED** [review]

**Review by mewim — APPROVED** [review]

**Pull request merged**